### ISSUE TITLE
Error while compiling with fresh Ubuntu Desktop V14 Desktop

### ISSUE THREAD

**Issue by admazzola** [PistonDevelopers/conrod#228]
Updating git repository `https://github.com/PistonDevelopers/current`
    Updating git repository `https://github.com/PistonDevelopers/texture`
    Updating git repository `https://github.com/PistonDevelopers/read_color`
    Updating git repository `https://github.com/PistonDevelopers/rust-sdl2`
    Updating git repository `https://github.com/pistondevelopers/shader_version`
    Updating git repository `https://github.com/bjz/gl-rs`
    Updating git repository `https://github.com/pistondevelopers/window`
    Updating git repository `https://github.com/netvl/xml-rs`
    Updating git repository `https://github.com/pistondevelopers/event_loop`
    Updating git repository `https://github.com/rust-lang/time`
    Updating registry `https://github.com/rust-lang/crates.io-index`
    Updating git repository `https://github.com/PistonDevelopers/freetype-sys`
    Updating git repository `https://github.com/pistondevelopers/image`
 Downloading log v0.1.2
 Downloading regex v0.1.3
 Downloading pkg-config v0.1.2
 Downloading libz-sys v0.1.0
 Downloading gcc v0.1.2
   Compiling xml-rs v0.1.4 (https://github.com/netvl/xml-rs#8821a34a)
   Compiling regex v0.1.3
/home/andy/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-0.1.3/src/re.rs             :544:21: 544:66 error: use of undeclared type name `std::slice::Items`
/home/andy/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-0.1.3/src/re.rs             :544     NamesIterNative(::std::slice::Items<'a, Option<&'static str>>),
                                                                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/andy/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-0.1.3/src/re.rs             :545:22: 545:61 error: use of undeclared type name `std::slice::Items`
/home/andy/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-0.1.3/src/re.rs             :545     NamesIterDynamic(::std::slice::Items<'a, Option<String>>)
                                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to 2 previous errors
   Compiling khronos_api v0.0.3 (https://github.com/bjz/gl-rs#eb8d6f6b)
   Compiling shader_version v0.0.0 (https://github.com/pistondevelopers/shader_v             ersion#d06e4472)
   Compiling pistoncore-input v0.0.2 (https://github.com/PistonDevelopers/input/             #887d85ec)
src/keyboard.rs:16:27: 16:36 warning: deriving(Encodable) is deprecated in favor              of deriving(RustcEncodable)
src/keyboard.rs:16     #[deriving(Decodable, Encodable)]
                                             ^~~~~~~~~
src/keyboard.rs:16:16: 16:25 warning: deriving(Decodable) is deprecated in favor              of deriving(RustcDecodable)
src/keyboard.rs:16     #[deriving(Decodable, Encodable)]
                                  ^~~~~~~~~
src/keyboard.rs:85:36: 85:45 warning: deriving(Encodable) is deprecated in favor              of deriving(RustcEncodable)
src/keyboard.rs:85 #[deriving(Copy, Clone, Decodable, Encodable, Show)]
                                                      ^~~~~~~~~
src/keyboard.rs:85:25: 85:34 warning: deriving(Decodable) is deprecated in favor              of deriving(RustcDecodable)
src/keyboard.rs:85 #[deriving(Copy, Clone, Decodable, Encodable, Show)]
                                           ^~~~~~~~~
src/mouse.rs:5:36: 5:45 warning: deriving(Encodable) is deprecated in favor of d             eriving(RustcEncodable)
src/mouse.rs:5 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq,
                                                  ^~~~~~~~~
src/mouse.rs:5:25: 5:34 warning: deriving(Decodable) is deprecated in favor of d             eriving(RustcDecodable)
src/mouse.rs:5 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq,
                                       ^~~~~~~~~
src/lib.rs:18:36: 18:45 warning: deriving(Encodable) is deprecated in favor of d             eriving(RustcEncodable)
src/lib.rs:18 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq, Eq, Show)             ]
                                                 ^~~~~~~~~
src/lib.rs:18:25: 18:34 warning: deriving(Decodable) is deprecated in favor of d             eriving(RustcDecodable)
src/lib.rs:18 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq, Eq, Show)             ]
                                      ^~~~~~~~~
src/lib.rs:27:36: 27:45 warning: deriving(Encodable) is deprecated in favor of d             eriving(RustcEncodable)
src/lib.rs:27 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq, Show)]
                                                 ^~~~~~~~~
src/lib.rs:27:25: 27:34 warning: deriving(Decodable) is deprecated in favor of d             eriving(RustcDecodable)
src/lib.rs:27 #[deriving(Copy, Clone, Decodable, Encodable, PartialEq, Show)]
                                      ^~~~~~~~~
src/lib.rs:38:30: 38:39 warning: deriving(Encodable) is deprecated in favor of d             eriving(RustcEncodable)
src/lib.rs:38 #[deriving(Clone, Decodable, Encodable, PartialEq, Show)]
                                           ^~~~~~~~~
src/lib.rs:38:19: 38:28 warning: deriving(Decodable) is deprecated in favor of d             eriving(RustcDecodable)
src/lib.rs:38 #[deriving(Clone, Decodable, Encodable, PartialEq, Show)]
                                ^~~~~~~~~
   Compiling pkg-config v0.1.2
   Compiling read_color v0.0.1 (https://github.com/PistonDevelopers/read_color#3             341ae05)
   Compiling gl_common v0.0.3 (https://github.com/bjz/gl-rs#eb8d6f6b)
   Compiling pistoncore-current v0.0.3 (https://github.com/PistonDevelopers/curr             ent#1b51b7d2)
src/current.rs:4:35: 4:43 error: unresolved import `std::collections::hash_map::             Occupied`. There is no `Occupied` in `std::collections::hash_map`
src/current.rs:4 use std::collections::hash_map::{ Occupied, Vacant };
                                                   ^~~~~~~~
src/current.rs:4:45: 4:51 error: unresolved import `std::collections::hash_map::             Vacant`. There is no `Vacant` in `std::collections::hash_map`
src/current.rs:4 use std::collections::hash_map::{ Occupied, Vacant };
                                                             ^~~~~~
error: aborting due to 2 previous errors
   Compiling vecmath v0.0.1 (https://github.com/PistonDevelopers/vecmath.git#238             30f2b)
   Compiling gcc v0.1.2
   Compiling image v0.2.0-alpha.4 (https://github.com/pistondevelopers/image#d00             71beb)
   Compiling piston-texture v0.0.1 (https://github.com/PistonDevelopers/texture#             0380876c)
Build failed, waiting for other jobs to finish...
src/png/decoder.rs:595:21: 595:33 error: mismatched types: expected `core::resul             t::Result<&str, core::str::Utf8Error>`, found `core::option::Option<_>` (expecte             d enum core::result::Result, found enum core::option::Option)
src/png/decoder.rs:595                     Some("IDAT") => (),
                                           ^~~~~~~~~~~~
error: aborting due to previous error
Could not compile `regex`.

**Comment by davekong** [comment]
After updating is compilation still failing for you?

**Comment by admazzola** [comment]
The compile failed and then I updated rust with ' curl -s
https://static.rust-lang.org/rustup.sh | sudo sh' and ran 'cargo update'
and tried again and then BOOM it worked :)  I got a massive amount of
errors like 'use of unstable item' right at the end but it compiled.  Also
this is the first time I ever got a decent sized rust project to compile
successfully with cargo :D (granted ive only screwed around with rust for
like 3 hours so far. ever.)   I develop indie games in Java Monkey Engine
3.  Rust looks like a cool improvement over Java.  I prob wont make the
switch for a long time but piston looks cool!  Im not a fan of game GUI
libraries, I usually roll my own in the end because many of them get too
high level which makes things easier, but restricts a lot of things. ( This
lib is a nightmare: https://github.com/void256/nifty-gui)    Anyways... yay!
